android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListViewTemplate.java

```diff
@@ -222,27 +223,35 @@ public DataItem getRootItem() {
 		return rootItem;
 	}
 
+	/**
+	 * 
+	 * @param data dictionary holding properties for template
+	 * @param update if true update entry else copy non-null values into data parameter
+	 */
 	public void updateOrMergeWithDefaultProperties(KrollDict data, boolean update) {
 		for (String binding: data.keySet()) {
 			DataItem dataItem = dataItems.get(binding);
 			if (dataItem == null) continue;
 
 			KrollDict defaultProps = dataItem.getDefaultProperties();
-			KrollDict props = new KrollDict((HashMap)data.get(binding));
+			KrollDict newProps = new KrollDict((HashMap) data.get(binding));
 			if (defaultProps != null) {
 				if (update) {
 					//update default properties
-					Set<String> existingKeys = defaultProps.keySet();
-					for (String key:  props.keySet()) {
-						if (!existingKeys.contains(key)) {
+					Set<String> defaultPropsKeys = defaultProps.keySet();
+					for (String key:  newProps.keySet()) {
+						if (!defaultPropsKeys.contains(key)) {
 							defaultProps.put(key, null);
 						}
 					}
 				} else {
 					//merge default properties with new properties and update data
-					HashMap<String, Object> newData = ((HashMap<String, Object>)defaultProps.clone());
-					newData.putAll(props);
-					data.put(binding, newData);
+					HashMap<String, Object> newData = ((HashMap<String, Object>) defaultProps.clone());
+					for (Map.Entry<String, Object> entry : newProps.entrySet()) {
+						if (entry.getValue() != null) {
+							newData.put(entry.getKey(), entry.getValue());
+						}
+					}
```

Comment:
Your code change in the "else" statement here no longer updates the "data" argument. It's making changes to a temporary "newData" local and then throwing it away.